lib/cdo/profanity_filter.rb

```diff
@@ -0,0 +1,33 @@
+require 'cdo/web_purify'
+
+class ProfanityFilter
+  # List of words that should be allowed only in specific languages.
+  #
+  # Entries are in the format
+  #   foobar: %w(en es)
+  # Where the key "foobar" is the word in question,
+  # and the value is a list of languages that should allow that word.
+  #
+  # Words in this list will be blocked for all languages _except_ for those
+  # listed along with the word.
+  # Words in this list should be _unblocked_ on WebPurify, since we are
+  # handling them with our custom code, here.
+  LANGUAGE_SPECIFIC_ALLOWLIST = {
+    fu: %w(it), # past-tense "to be" in Italian
+    fick: %w(sv) # "got" in Swedish
+  }
+
+  # Look for profanity in a given text, return the first expletive found
+  # or nil if no profanity is found.
+  #
+  # @param [String] text to check for profanity
+  # @param [String] language_code a two-character ISO 639-1 language code
+  def self.find_potential_profanity(text, language_code)
+    LANGUAGE_SPECIFIC_ALLOWLIST.each do |word, languages|
+      next if languages.include? language_code
+      r = Regexp.new "\\b#{word}\\b", Regexp::IGNORECASE
+      return word.to_s if r =~ text
+    end
+    WebPurify.find_potential_profanity(text, ['en', language_code])
+  end
```

Comment:
An alternative I briefly considered and am now thinking might be better: For each word, if we _are_ using one of the allowed languages, strip the word out of the text before we send it to WebPurify.  Continue letting WebPurify make the go-no-go call.

- Pro: We only configure words here, we don't also have to unblock them in WebPurify.
- Pro: Our configuration has a more targeted effect on specific languages, will not necessarily block the word in question for all _other_ languages.
- Con: It seems less "correct" to modify the text this way before sending it along - I don't know if this would impact WebPurify's effectiveness (if they're using n-grams, for example).
- Con: We aren't shortcutting any WebPurify API calls.

Thoughts?

Reply:
sounds interesting, but could get weird -- webpurify also looks for things like addresses and phone numbers, right? so if you have something like `555 fu 1212` and the `fu` gets stripped, webpurify might complain about the now-PII-like number. unsure if this will happen in practice, but modifying what we send is making me worry.

Reply:
We've got WebPurify's PII filtering turned off right now:

![image](https://user-images.githubusercontent.com/1615761/58842874-efb3e680-8624-11e9-99a2-1a04fe37690f.png)

Because we check those ourselves before we contact WebPurify:

https://github.com/code-dot-org/code-dot-org/blob/4ba0d0adbb90cd1845d7253e8ca6e31ec63eb00a/lib/cdo/share_filtering.rb#L32-L42

But I share your concern.

Reply:
IMHO the number of problematic words has been fairly small in practice, so I think requiring the manual step of adding exceptions via WebPurify is a perfectly reasonable and scrappy solution.